packages/editor/src/lib/components/Shape.tsx

```diff
@@ -71,7 +69,13 @@ export const Shape = memo(function Shape({
 			}
 
 			// Page transform
-			const transform = Mat.toCssString(editor.getShapePageTransform(id))
+			const pageTransform = editor.getShapePageTransform(shape).clone()
+			const bounds = editor.getShapeGeometry(shape).bounds
+			if (isCulled) {
```

Comment:
We have to transform the position according to the bounds, just like we did in the `CullingShape` below